File: library/src/main/java/de/nenick/espressomacchiato/elements/EspTextView.java
package de.nenick.espressomacchiato.elements;

import android.view.View;

import org.hamcrest.Matcher;

import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

public class EspTextView extends EspView {

    public static EspTextView byId(int resourceId) {
        return new EspTextView(resourceId);
    }

    public static EspTextView byText(String text) {
        return new EspTextView(withText(text));
    }

    public EspTextView(int resourceId) {
        super(resourceId);
    }

    public EspTextView(Matcher<View> base) {
        super(base);
    }

    public void assertTextIs(String expected) {
        findView().check(matches(withText(expected)));
    }
}

File: esp-core/src/main/java/de/nenick/espressomacchiato/elements/basics/EspTextView.kt
package de.nenick.espressomacchiato.elements.basics

import android.view.View
import androidx.annotation.IdRes
import androidx.test.espresso.Root
import androidx.test.espresso.matcher.ViewMatchers
import de.nenick.espressomacchiato.espresso.view.TextAssertions
import de.nenick.espressomacchiato.internals.OpenForExtensions
import org.hamcrest.Matcher

@OpenForExtensions
class EspTextView(
        viewMatcher: Matcher<View>,
        rootMatchers: Matcher<Root>? = null,
        interactions: EspTextView.() -> Unit = {}
) :
        EspView(viewMatcher, rootMatchers, interactions as EspView.() -> Unit),
        TextAssertions {

    constructor(@IdRes viewId: Int = 0, interactions: EspTextView.() -> Unit = {})
            : this(viewId, null, interactions)

    constructor(@IdRes viewId: Int = 0, rootMatchers: Matcher<Root>? = null, interactions: EspTextView.() -> Unit = {})
            : this(ViewMatchers.withId(viewId), rootMatchers, interactions)
}